### PR TITLE
[SPARK-43428] Move some class utils to common/utils

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

This PR moves some commonly used class loader/reflection utils to common/utils.

### Why are the changes needed?

Reduce the required dependency on Spark core.

### Does this PR introduce _any_ user-facing change?

NO
### How was this patch tested?

Existing UT

### CODE REVIEW

**Comment by amaliujia** [comment]
@hvanhovell @cloud-fan

File: common/utils/src/main/scala/org/apache/spark/util/ClassUtils.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.util
+
+object SparkClassUtils {
```
[the remark this answers is not in the document]
Comment (amaliujia) [reply common/utils/src/main/scala/org/apache/spark/util/ClassUtils.scala]: ah yeah updated.

**Review by hvanhovell — APPROVED** [review]
LGTM

File: core/src/main/scala/org/apache/spark/util/Utils.scala [hunk 2]
```diff
@@ -197,7 +197,7 @@ private[spark] object Utils extends Logging {
   /**
    * Get the ClassLoader which loaded Spark.
    */
-  def getSparkClassLoader: ClassLoader = getClass.getClassLoader
+  def getSparkClassLoader: ClassLoader = SparkClassUtils.getSparkClassLoader
```
Comment (hvanhovell) [inline core/src/main/scala/org/apache/spark/util/Utils.scala:200]: You implement `SparkClassUtils` as a trait, and make `Utils` implement that trait. That will save you from writing a lot of code like this.

**Comment by hvanhovell** [comment]
Merging.

File: core/src/main/scala/org/apache/spark/util/Utils.scala (hunk 2 above)
Comment (amaliujia) [reply core/src/main/scala/org/apache/spark/util/Utils.scala:200]: This is a good idea.

**Comment by LuciferYang** [comment]
A late question: Should `error/error-classes.json` also be moved to the `common-utils` module?

**Pull request closed without merging**